src/ort_genai_c.h

```diff
@@ -508,6 +508,22 @@ OGA_EXPORT OgaResult* OGA_API_CALL OgaProcessorDecode(const OgaMultiModalProcess
  */
 OGA_EXPORT OgaResult* OGA_API_CALL OgaTokenizerApplyChatTemplate(const OgaTokenizer*, const char* template_str, const char* messages, bool add_generation_prompt, const char** out_string);
 
+/**
+ * @brief Applies a chat template to input messages and tokenizes them
+ *
+ * This function processes the specified template with the provided input using the
+ * tokenizer, and outputs the tokenized input ids of the resulting string. Optionally,
+ * it can include a generation prompt in the output.
+ *
+ * \param[in] tokenizer OgaTokenizer used for template processing.
+ * \param[in] template_str Null-terminated string representing the chat template. Use nullptr to fall back to the default chat template from the tokenizer config.
+ * \param[in] messages Null-terminated string containing the input messages to be processed.
+ * \param[in] add_generation_prompt Indicates whether to add a generation prompt to the output.
+ * \param[out] out_ids Pointer to where the output will be stored. The returned pointer must be freed with OgaDestroyString
+ * \return OgaResult* containing the error message if the function fails
+ */
+OGA_EXPORT OgaResult* OGA_API_CALL OgaTokenizerApplyChatTemplateTokenize(const OgaTokenizer*, const char* template_str, const char* messages, bool add_generation_prompt, OgaSequences* sequences);
+
```

Comment:
As we discussed offline, this can be removed. Just need the one to apply the chat template. Tokenization will use the existing API to keep it simple (just one way to do it vs two)

Reply:
yep, done!